### PR TITLE
Resilient witness table emission cleanup

### PR DESCRIPTION
**Pull request by slavapestov**
I found some of the code in GenProto.cpp hard to follow while working on https://github.com/apple/swift/pull/22737. Now that the immediate bug fix is done here is a larger cleanup. Should be mostly NFC.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: test/IRGen/protocol_conformance_records.swift [hunk 1]
```diff
@@ -136,7 +157,7 @@ extension Int : OtherResilientProtocol { }
 // -- witness table pattern
 // CHECK-SAME:           @"$s28protocol_conformance_records9DependentVyxGAA9AssociateAAWp"
 // -- flags
-// CHECK-SAME:           i32 0
+// CHECK-SAME:           i32 131072,
```
Comment (slavapestov) [inline test/IRGen/protocol_conformance_records.swift]: Note that this is not a change in test output -- the test was matching the wrong "i32 0"

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
[apple/swift-lldb#1314](https://github.com/apple/swift-lldb/pull/1314)
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
This PR is cursed. Now I'm seeing

```
/home/buildnode/jenkins/workspace/swift-PR-Linux-smoke-test/branch-master/swift-corelibs-libdispatch/src/swift/Dispatch.swift:13:19: error: no such module 'Dispatch'
00:38:50.035 @_exported import Dispatch
00:38:50.035                   ^
```

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by shahmishal** [comment]
cc/ @millenomi have you seen this issue?

**Comment by shahmishal** [comment]
@millenomi Sorry, this is dispatch issue.

**Comment by ktopley-apple** [comment]
> /cc @ktopley-apple have you seen this issue before?

No. Which build was this? I looked at #11255 and I did not see any dispatch errors.

**Comment by slavapestov** [comment]
Now we're failing with:

```
01:25:45.115 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift-corelibs-foundation/TestFoundation/TestURLSession.swift:170: error: TestURLSession.test_dataTaskWithHttpInputStream : XCTAssertEqual failed: ("1316 bytes") is not equal to ("1323 bytes") - Response Data and Data is not equal
01:25:45.115 Test Case 'TestURLSession.test_dataTaskWithHttpInputStream' failed (0.003 seconds)
```

@shahmishal Can we disable Foundation tests in PR testing for now?

**Comment by shahmishal** [comment]
Yes, we should disable this test. cc: @millenomi

**Comment by slavapestov** [comment]
https://github.com/apple/swift-corelibs-foundation/pull/1939
@swift-ci Please smoke test Linux
